### PR TITLE
[Feature] add feature mapping to the SiriusAdapter

### PR DESCRIPTION
**Pull request by oliveralka**
FeatureFinderMetabo:
1. change output format of masstrace_intensity
2. add output masstrace centroid rt and centroid mz
3. fix the corresponding tests

SiriusAdapter:
1. add feature mapping to the SiriusAdapter (ms2 to feature in the minimal distance).
2. refactor function to extract isotope information (C13) if no feature is available
3. extract masstrace ms1 information in case of feature is available
4. write the feature_id to the corresponding output
5. some smaller changes 
6. fix the test (the file for test_3 was replaced with a better match)
7. update the Sirius internal .ms format

edit:
There was a problem with "atoi" and the sort functionality with different compilers (gcc and clang). Refactor of the sort functionality (without the use of "atoi").

### CODE REVIEW

**Comment by oliveralka** [comment]
rebuild jenkins

File: src/openms/source/FILTERING/DATAREDUCTION/FeatureFindingMetabo.cpp [hunk 1]
```diff
@@ -910,12 +931,11 @@ namespace OpenMS
       // store isotope intensities
       std::vector<double> all_ints(feat_hypos[hypo_idx].getAllIntensities(use_smoothed_intensities_));
       f.setMetaValue("num_of_masstraces", all_ints.size());
-      for (Size int_idx = 0; int_idx < all_ints.size(); ++int_idx)
-      {
-        f.setMetaValue("masstrace_intensity_" + String(int_idx), all_ints[int_idx]);
-      }
       if (report_convex_hulls_) f.setConvexHulls(feat_hypos[hypo_idx].getConvexHulls());
       f.setOverallQuality(feat_hypos[hypo_idx].getScore());
+      f.setMetaValue("masstrace_intensity", all_ints);
```
[the remark this answers is not in the document]
Comment (oliveralka) [reply src/openms/source/FILTERING/DATAREDUCTION/FeatureFindingMetabo.cpp]: Will do. 
Seems to be used in four cases:
https://github.com/OpenMS/OpenMS/search?q=masstrace_intensity_&unscoped_q=masstrace_intensity_&type=Code

File: src/utils/SiriusAdapter.cpp [hunk 2]
```diff
@@ -381,7 +432,7 @@ class TOPPSiriusAdapter :
       csifile.store(out_csifingerid, csi_result);
     }
 
-    //clean tmp directory if debug level < 2
+    // clean tmp directory if debug level < 2
     if (debug_level_ >= 2)
```
Comment (jpfeuffer) [inline src/utils/SiriusAdapter.cpp]: Consider using the functions in TOPPBase. E.g. makeAutoRemoveTempDirectory that will be removed if they go out of scope.

Comment (oliveralka) [reply src/utils/SiriusAdapter.cpp]: Will add that next time.

**Comment by oliveralka** [comment]
REFACTOR:
Mapping of MS2 spectra to feature in minimum distance -> FeatureMapping.cpp/FeatureMapping.h
SiriusMSConverter can provide an additional format for Sirius:
Multiple MS1 (precursor spectra) and MS2 for one compound
Can use four different modes depending on input parameters
1. mzML only
2. mzML and featureXML (can use adduct information for all annotated features)
3. mzML, featureXML and "-feature_only" (uses feature/adduct information only)
4. mzML, feautreXML, "-feautre_only", "-filter_by_num_masstraces" (reduces the feature space based on number of masstraces)
FIX:
Corresponding SiriusAdapter, AccurateMassSearch

Important:
Due to changes of the FeatureFinderMetabo Output format - from this point on AccurateMassSearch and SiriusAdapter will only work with new FFM.featureXML files.

File: src/openms/include/OpenMS/ANALYSIS/ID/SiriusMSConverter.h [hunk 3]
```diff
@@ -34,12 +34,12 @@
 
 #pragma once
 
-#include <OpenMS/DATASTRUCTURES/String.h>
-#include <OpenMS/DATASTRUCTURES/StringListUtils.h>
 #include <OpenMS/KERNEL/StandardTypes.h>
```
Comment (hroest) [inline src/openms/include/OpenMS/ANALYSIS/ID/SiriusMSConverter.h]: I dont think you need any of these includes except `StandardTypes.h` - you can forward declare the rest, right?

Comment (oliveralka) [reply src/openms/include/OpenMS/ANALYSIS/ID/SiriusMSConverter.h]: That is right!

**Comment by oliveralka** [comment]
I tried to get the header slimmer, but I am not sure how to further check if the includes are needed or how to optimally use a forward declaration.

**Comment by hroest** [comment]
> I tried to get the header slimmer, but I am not sure how to further check if the includes are needed or how to optimally use a forward declaration.

we dont really have a good / automated way in general in OpenMS, so its educated guesses :-)

File: src/openms/include/OpenMS/FORMAT/DATAACCESS/CsiFingerIdMzTabWriter.h [hunk 4]
```diff
@@ -62,8 +62,8 @@ namespace OpenMS
 
           struct CsiAdapterIdentification
           {
-            OpenMS::String scan_index;
```
[the remark this answers is not in the document]
Comment (oliveralka) [reply src/openms/include/OpenMS/FORMAT/DATAACCESS/CsiFingerIdMzTabWriter.h]: done

**Pull request merged**